flytekit/core/launch_plan.py

```diff
@@ -93,6 +102,61 @@ def create(
         cls.CACHE[name] = lp
         return lp
 
+    @classmethod
+    def get_or_create(
+        cls,
+        workflow: _annotated_workflow.WorkflowBase,
+        name: Optional[str] = None,
+        default_inputs: Dict[str, Any] = None,
+        fixed_inputs: Dict[str, Any] = None,
+        schedule: _schedule_model.Schedule = None,
```

Comment:
should we put all of this 
`schedule, notifications, auth_role` etc in a dataclass? something like meta etc
as we keep adding things here?

Reply:
Maybe, but then the user has to make one more object to make this.  Can we hold off until we have one more thing?  It makes sense I guess if they're creating fifteen launch plans, and set all three settings with the same value - then they can create the meta once and use fifteen times, but i feel like that's not a common use case.

Reply:
if you do later, then it becomes a breaking change. I guess you are right though, this list since its optional can keep on increasing. 

Just that your first check for nil, has to keep evolving as well